### PR TITLE
test: functional suite for approve flow for permissioned user and random one

### PR DESCRIPTION
**Pull request by petrovska-petro**
handles #3

### CODE REVIEW

File: tests/revoke_module/test_approve_action.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+from brownie import reverts
+
+
+def test_approve(module, safe, deployer, token, rug_puller, module_set_up):
+    assert token.allowance(safe, rug_puller) > 0
+    tx = module.approve(safe, token, rug_puller, {"from": deployer})
```
Comment (gosuto-inzasheru) [inline tests/revoke_module/test_approve_action.py]: calling this function `approve` is actually confusing; proposing `revoke`

File: tests/revoke_module/conftest.py [hunk 2]
```diff
@@ -17,7 +14,20 @@ def module(deployer):
...
-    fake_token = oz.ERC20.deploy("rugPullToken", "NGMI", {"from": deployer})
+def token(deployer):
+    fake_token = TokenErc20.deploy(
+        "rugPullToken", "NGMI", deployer, 100e18, {"from": deployer}
+    )
     return fake_token
+
+
+@pytest.fixture()
+def module_set_up(safe, module, token, rug_puller):
```
Comment (gosuto-inzasheru) [inline tests/revoke_module/conftest.py]: why separate this from `module`? can be integrated into one imo

File: contracts/modules/RevokeModule.sol
Comment (gosuto-inzasheru) [inline contracts/modules/RevokeModule.sol:1]: wondering if the signer actually needs to pass `safe` to the revoke func here? safer to always refer to `address(this)` no? also saves the signer one address lookup, which means quicker calling.

File: tests/revoke_module/test_approve_action.py (hunk 1 above)
Comment (petrovska-petro) [reply tests/revoke_module/test_approve_action.py]: sorted in fa03684d6593df53c8a39a0b0b5c2ccb7b4422fd

File: tests/revoke_module/conftest.py (hunk 2 above)
Comment (petrovska-petro) [reply tests/revoke_module/conftest.py]: sorted in fa03684d6593df53c8a39a0b0b5c2ccb7b4422fd

File: contracts/modules/RevokeModule.sol
Comment (petrovska-petro) [reply contracts/modules/RevokeModule.sol:1]: solution encapsulated in #16

**Review by gosuto-inzasheru — APPROVED** [review]

**Pull request merged**